fix(itunes): sms auth [NATIVE-265]

flub opened the pull request:
An attempt to let users run less into the problem with sms auth vs normal auth.

- [x] Remove useSms from backend, no longer accept it in requests
- [x] Remove useSms from UI, it no longer sends this
- [x] UI displays `itunes-sms-blocked-error` in a translated string instead of generic "an error occurred"

Fixes SENTRY-S5G

Commit:
fix(itunes): Improve handling of SMS request failures

The client was changing the useSms flag regardless of whether the SMS
authentication code was requested.  This only sets this variable if
the request to the backend was successful.

Commit:
Ignore the UI's idea of useSms flag

We already have the state in the itunes client, we may as well not
bother asking the client about this.

flub commented on static/app/components/modals/debugFileCustomRepository/appStoreConnect/index.tsx:
```diff
@@ -361,10 +357,11 @@ function AppStoreConnect({
         }
       );
-
+      setUseSms(true);
```
In the 2nd commit I changed the python code to entirely ignore this flag from the client, we already get this state via the sessionContext anyway.  So we could decide to remove it entirely out of the UI code and also no longer accept it on the backend API?

flub replied:
anyway, one bug was that this was set regardless of whether the API call succeeded.  hence i move this.

relaxolotl replied:
is `useSms` something that may be included in the response payload? is it possible for the request to be successful and the response to contain `useSms: false` with some sort of hint pointing you to a trusted device?

priscilawebdev replied:
> anyway, one bug was that this was set regardless of whether the API call succeeded. hence i move this.

this is strange..did you manage to reproduce the bug?

flub replied:
user hits sms endpoint, sms endpoint returns error, client has useSms set but it actually isn't.  no not fully reproduced, but i understand how this can occur.

flub commented on static/app/components/modals/debugFileCustomRepository/appStoreConnect/index.tsx:
```diff
@@ -361,10 +357,11 @@ function AppStoreConnect({
       setSessionContext(response.sessionContext);
       addSuccessMessage(t("We've sent a SMS code to your phone"));
     } catch {
+      // TODO: use the message from the itunes-sms-blocked-error
```
@priscilawebdev It it possible to change the message based on the response here?  We return a special "blocked" response from the backend with `code = itunes-sms-blocked-error` and a message: https://github.com/getsentry/sentry/blob/9423b9d13fe68636db0287b20c20b9d5e20d9fb8/src/sentry/api/exceptions.py#L118-L121. It would be really helpful to show this to the user because in that case "please try again" is the wrong advice and will only make things worse.

priscilawebdev replied:
sure it is possible but I would have anyway to use convert it to translations on the client side

priscilawebdev replied:
https://github.com/getsentry/sentry/blob/master/static/app/views/organizationGroupDetails/grouping/errorMessage.tsx

Commit:
Completely remove useSms from /2fa/ endpoint

Since this is no longer use remove it from the protocol and no longer
accep the field in the JSON body.

priscilawebdev committed:
Remove useSms from UI, it no longer sends this

priscilawebdev committed:
Map request erros in the frontend

priscilawebdev committed:
cr feedback

priscilawebdev committed:
adjust error message format

priscilawebdev reviewed (approved):
tested locally and everything seems to work fine

priscilawebdev committed:
applied cr feedback

flub reviewed (commented):
LGTM, tested manually and everything works

Commit:
Merge branch 'master' into fix/itunes-sms-auth
